### ISSUE TITLE
💅 Misplaced highlight of `suppressions/incorrect` in JSX

### ISSUE THREAD

**Issue by Loskir** [biomejs/biome#6654]
### Environment information

```bash
Not applicable
```

### Rule name

suppressions/incorrect

### Playground link

[playground](https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAQwBvAG0AcABvAG4AZQBuAHQAKAApACAAewAKACAAIAByAGUAdAB1AHIAbgAgACgAPAA%2BAAoAIAAgACAAIAB7AC8AKgAqACAAYgBpAG8AbQBlAC0AaQBnAG4AbwByAGUAIABsAGkAbgB0AC8AYwBvAG0AcABsAGUAeABpAHQAeQAvAG4AbwBVAHMAZQBsAGUAcwBzAEYAcgBhAGcAbQBlAG4AdABzADoAIAA8AGUAeABwAGwAYQBuAGEAdABpAG8AbgA%2BACAAKgAvAH0ACgAgACAAIAAgADwAPgA8AC8APgAKACAAIAA8AC8APgApAAoAfQA%3D&ruleDomains.react=recommended)

### Expected result

Should highlight `<explanation>`

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
